drivers/docker/driver.go

```diff
@@ -860,9 +860,14 @@ func (d *Driver) createContainerConfig(task *drivers.TaskConfig, driverConfig *T
 
 	hostConfig.ReadonlyRootfs = driverConfig.ReadonlyRootfs
 
+	// set the docker network mode
 	hostConfig.NetworkMode = driverConfig.NetworkMode
+
+	// if the driver config does not specify a network mode then try to use the
+	// shared alloc network
 	if hostConfig.NetworkMode == "" {
-		if task.NetworkIsolation.Path != "" {
+		if task.NetworkIsolation != nil && task.NetworkIsolation.Path != "" {
```

Comment:
I like helpers for these cases as it's a clearer signal to developers that a nil check is necessary. Something like:

```go
if task.NetworkIsolation.HasPath()
```

`Deployment` has a bunch of helpers like this. I don't think it's always necessary, so not a blocker (can also be done later).